presto-spi/src/main/java/com/facebook/presto/spi/StandardErrorCode.java

```diff
@@ -63,6 +63,7 @@
     QUERY_HAS_TOO_MANY_STAGES(0x0000_0028, USER_ERROR),
     INVALID_SPATIAL_PARTITIONING(0x0000_0029, USER_ERROR),
     INVALID_ANALYZE_PROPERTY(0x0000_002A, USER_ERROR),
+    EXCEEDED_MAX_ARRAY_SIZE(0x0000_002B, USER_ERROR),
```

Comment:
@yingsu00 : I agree an error code like `EXCEEDED_MAX_ARRAY_SIZE` sound too specific -- we might also want to restrict map or varchar size in the future. But `EXCEEDED_FUNCTION_MEMORY_LIMIT` doesn't sound like for this purpose as well.

What about `CELL_TOO_LARGE`? -- basically it complaints the single array result from `array_agg` is too large. cc @highker , @rschlussel

Reply:
> @yingsu00 : I agree an error code like `EXCEEDED_MAX_ARRAY_SIZE` sound too specific -- we might also want to restrict map or varchar size in the future. But `EXCEEDED_FUNCTION_MEMORY_LIMIT` doesn't sound like for this purpose as well.

I saw `EXCEEDED_FUNCTION_MEMORY_LIMIT` was only used in TypedSet#addNewElement which was mostly used aggregation functions. It is for very similar situation where a single element is too large to be added to destination BlockBuilder. If we're going to introduce new error code, maybe we should unify the usages.

Reply:
Leaning towards ```EXCEEDED_FUNCTION_MEMORY_LIMIT``` as well since it is for a similar use case, and if we don't want to add very specific error codes.

Reply:
@sujay-jain @yinghai : No it's not the memory used by function. `EXCEEDED_FUNCTION_MEMORY_LIMIT` should be used when some function internal state exceeds memory limit, such as the internal hash table. In this case, the aggregation internal memory usage is correctly tracked and doesn't exceed. It's the final produced result too large.

Reply:
@sujay-jain : Please don't open discussion as "resolved" as it makes difficult to track.

Reply:
> @sujay-jain : Please don't open discussion as "resolved" as it makes difficult to track.
sorry about that! wasn't sure about the best flow :)